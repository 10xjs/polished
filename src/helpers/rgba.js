// @flow

import rgb from './rgb'

type RgbaColor = {
  red: number,
  green: number,
  blue: number,
  alpha: number,
}

function rgba(value: RgbaColor | number, green?: number, blue?: number, alpha?: number): string {
  if (typeof value === 'number' &&
      typeof green === 'number' &&
      typeof blue === 'number' &&
      typeof alpha === 'number') {
    return alpha === 255 ? rgb(value, green, blue) : `rgba(${value},${green},${blue},${alpha})`
  } else if (typeof value === 'object' && green === undefined && blue === undefined && alpha === undefined) {
    return value.alpha === 255
      ? rgb(value.red, value.green, value.blue)
      : `rgba(${value.red},${value.green},${value.blue},${value.alpha})`
  }

  throw new Error('Passed invalid arguments to rgba, please pass multiple numbers e.g. rgb(255, 205, 100, 180) or an object e.g. rgb({ red: 255, green: 205, blue: 100, alpha: 180 }).')
}

export default rgba
