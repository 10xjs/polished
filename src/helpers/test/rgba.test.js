// @flow
import rgba from '../rgba'

describe('rgb', () => {
  it('should convert multiple numbers to a rgba string', () => {
    expect({ background: rgba(255, 205, 100, 180) }).toMatchSnapshot()
  })

  it('should convert multiple numbers with full opacity to a hex color', () => {
    expect({ background: rgba(255, 205, 100, 255) }).toMatchSnapshot()
  })

  it('should convert a rgba object to a rgba string', () => {
    expect({ background: rgba({ red: 255, green: 205, blue: 100, alpha: 180 }) }).toMatchSnapshot()
  })

  it('should convert a rgba object with full opacity to a hex color', () => {
    expect({ background: rgba({ red: 255, green: 205, blue: 100, alpha: 255 }) }).toMatchSnapshot()
  })

  it('should convert a rgba object with full opacity to a reduced hex color', () => {
    expect({ background: rgba({ red: 255, green: 255, blue: 255, alpha: 255 }) }).toMatchSnapshot()
  })

  it('should throw an error if an object and multiple arguments are passed', () => {
    expect(() => ({ background: rgba({ red: 255, green: 1, blue: 1, alpha: 180 }, 250, 100) }))
      .toThrow('Passed invalid arguments to rgba, please pass multiple numbers e.g. rgb(255, 205, 100, 180) or an object e.g. rgb({ red: 255, green: 205, blue: 100, alpha: 180 }).')
  })
})
