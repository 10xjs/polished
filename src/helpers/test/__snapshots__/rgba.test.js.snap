exports[`rgb should convert a rgba object to a hex color 1`] = `
Object {
  "background": "rgba(255,205,100,180)",
}
`;

exports[`rgb should convert multiple numbers to a rgba string 1`] = `
Object {
  "background": "rgba(255,205,100,180)",
}
`;
